src/bin/str.rs

```diff
@@ -54,30 +71,56 @@ impl fmt::Display for Constant {
 impl std::str::FromStr for Constant {
     type Err = ();
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match i64::from_str(s) {
-            Ok(i) => Ok(Self::Num(i)),
-            Err(_) => {
-                let s = s.trim_matches('"');
-                Ok(Self::Str(s.as_bytes().into()))
-            }
+        if let Ok(i) = i64::from_str(s) {
+            Ok(Self::Num(i))
+        } else {
+            Ok(Self::Str(SmallString::from_str(s.trim_matches('"'))))
```

Comment:
why is the `trim_matches` needed?

Reply:
that was in the original implementation ([here](https://github.com/uwplse/ruler/blob/main/src/bin/str.rs#L60)) so I just left it. I think it just removes leading/trailing quotes?